jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/main/java/org/jbpm/workbench/ks/integration/KieServerIntegration.java

```diff
@@ -165,18 +169,13 @@ public void onServerTemplateDeleted(@Observes ServerTemplateUpdated serverTempla
 
     public void onServerTemplateDeleted(@Observes ServerTemplateDeleted serverTemplateDeleted) {
         // remove all clients for this server template and its containers
-        Iterator<Map.Entry<String, KieServicesClient>> iterator = serverTemplatesClients.entrySet().iterator();
-        while (iterator.hasNext()) {
-            Map.Entry<String, KieServicesClient> entry = iterator.next();
-            if (entry.getKey().startsWith(serverTemplateDeleted.getServerTemplateId())) {
-                //KieServicesClient client = entry.getValue();
-                //client.close();
-                logger.debug("KieServerClient removed and closed for server template {}",
-                             entry.getKey());
+        serverTemplatesClients.computeIfPresent(serverTemplateDeleted.getServerTemplateId(),
```

Comment:
what's the reason of putting null into the map for given server template id instead of removing it?

Reply:
that's indeed the intention, removing instead of null. It happens that for _computeIfPresent_ returning null will actually remove the key;

Reply:
isn't it then less readable (and maybe even less efficient)? I mean calling it serverTemplatesClients.remove would be self explanatory...

Reply:
done